adapters/coap/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java

```diff
@@ -138,11 +138,14 @@ protected Future<RequestDeviceAndAuth> getPutRequestDeviceAndAuth(final CoapExch
         Optional.ofNullable(TracingSupportingHonoResource.getAuthenticatedDevice(exchange))
             .ifPresentOrElse(
                     authenticatedDevice -> {
-                        final String tenantFromUri = Optional.ofNullable(requestedResource.getTenantId())
+                        final String tenantId = Optional.ofNullable(requestedResource.getTenantId())
```

Comment:
IMHO we should also return a failed future (404) if the request URI does not contain a tenant ID as the CoAP user guide seems to require it for a PUT request, shouldn't we?

Reply:
A tenant ID is required for an _unauthenticated_ device request. This case is handled in the block starting on line 157.
There is already a 404 returned in this case (see also test method param `"false,/event//device_1,404",`).

Reply:
Oh yes, my bad, the PUT operation is also used by an authenticated gateway ...